rust/src/write.rs

```diff
@@ -32,6 +32,13 @@ enum WriteMode<W: Write + Seek> {
     Attachment(AttachmentWriter<CountingCrcWriter<W>>),
 }
 
+pub enum PrivateRecordKind {
```

Comment:
I think shifting to PrivateRecordOptions (maybe an enum set or a struct of bools) is the right move.